lib/Template/Document.pm

```diff
@@ -293,7 +292,7 @@ sub write_perl_file {
     my ($fh, $tmpfile);
     
     return $class->error("invalid filename: $file")
-        unless $file =~ /^(.+)$/s;
+        unless length $file;
```

Comment:
need to protect for older perl versions
`unless defined $file && length $file`

Reply:
The original code did not.